localstack/utils/aws/message_forwarding.py

```diff
@@ -47,7 +48,16 @@ def send_event_to_target(
             role_arn=role, service_principal=source_service, region_name=region
         )
     else:
-        clients = connect_to(region_name=region)
+        try:
+            from localstack_ext.config import ENFORCE_IAM
+
+            clients = (
+                connect_to(region_name=region)
+                if ENFORCE_IAM
+                else connect_to(aws_access_key_id=account_id, region_name=region)
+            )
+        except ImportError:
+            clients = connect_to(aws_access_key_id=account_id, region_name=region)
```

Comment:
Are you sure about introducing this circular dependency? For happy path, shouldn't it be ideal to pass the role?

Reply:
Passing the role param here was already discussed and approved earlier. The new fix of introducing the `ENFORCE_IAM` attribute accounts to handle the failing ext test cases as mentioned [here](https://github.com/localstack/localstack/pull/9784#issuecomment-1938155138). Waiting for @dfangl review on this.